### PR TITLE
Json handler implementation and bug fixes

### PR DESCRIPTION
**Pull request by mnamburi**

### CODE REVIEW

File: bigdime-data-handlers/src/main/java/io/bigdime/handler/hive/HiveMetaDataHandler.java [hunk 1]
```diff
@@ -61,16 +62,21 @@
 	private HiveDBManger hiveDBManager = null;
 	private HiveTableManger hiveTableManager = null;
 	private HivePartitionManger hivePartitionManager = null;
+
 	private static String hdfsScheme = null;
-	
-	@Autowired private MetadataStore metadataStore;
 
+	private static final String HIVE_SCHEME = "hive.uri.hdfs.scheme";
+	private static final String HIVE_DEFAULT_SCHEME = "hdfs://";
```
Comment (naithania) [inline bigdime-data-handlers/src/main/java/io/bigdime/handler/hive/HiveMetaDataHandler.java]: Do you think,we should have a constants in separate class ?

File: bigdime-commons/src/main/java/io/bigdime/core/commons/JsonHelper.java [hunk 2]
```diff
@@ -38,14 +38,71 @@ public boolean getBooleanProperty(final JsonNode node, final String key) {
 		}
 	}
 
+	/**
+	 * If the node is not  present return -1,
+	 * 
+	 * @param node
+	 * @param key
+	 * @return true if the node is present has the value as true, false
```
Comment (a-ninja) [inline bigdime-commons/src/main/java/io/bigdime/core/commons/JsonHelper.java]: Update javadoc

File: bigdime-commons/src/main/java/io/bigdime/core/commons/JsonHelper.java [hunk 3]
```diff
@@ -38,14 +38,71 @@ public boolean getBooleanProperty(final JsonNode node, final String key) {
...
+	 *         otherwise.
+	 */
+	public int getIntProperty(final JsonNode node, final String key) {
+		try {
+			String stringValue = getRequiredStringProperty(node, key);
+			return Integer.valueOf(stringValue);
+		} catch (IllegalArgumentException ex) {
+			return -1;
```
Comment (a-ninja) [inline bigdime-commons/src/main/java/io/bigdime/core/commons/JsonHelper.java]: I think it should actually throw the IllegalArgumentException.

**Comment by naithania** [comment]
looks good to me.

File: bigdime-data-handlers/src/main/java/io/bigdime/handler/hive/AdapterMetaDataHandler.java [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+
+@Component
+@Scope("prototype")
+/**
+ * This class responsible to load the schema from a file system to bigdime metastore.
+ * 
+ * @author mnamburi
+ *
+ */
+public class AdapterMetaDataHandler  extends AbstractHandler {
```
Comment (a-ninja) [inline bigdime-data-handlers/src/main/java/io/bigdime/handler/hive/AdapterMetaDataHandler.java]: Adaptor or Adapter?

File: bigdime-data-handlers/src/main/java/io/bigdime/handler/json/JsonMapperHandler.java [hunk 5]
```diff
@@ -0,0 +1,297 @@
...
+
+		try {
+			assginPartitionsFromRunTimeInfo(inputDescriptor.getEntityName(), String.valueOf(inputDescriptor.getPartition()));
+		} catch (RuntimeInfoStoreException e) {
+			throw new AdaptorConfigurationException(e);
+		}		
+		logger.debug("building handler", handlerPhase+"timestamp={} partition_name={}",timestamp,partition_name);
+	}
+	
+	public KafkaInputDescriptor getInputDescriptor(){
```
Comment (a-ninja) [inline bigdime-data-handlers/src/main/java/io/bigdime/handler/json/JsonMapperHandler.java]: is this method needed here?

File: bigdime-data-handlers/src/main/java/io/bigdime/handler/json/JsonMapperHandler.java [hunk 6]
```diff
@@ -0,0 +1,297 @@
...
+	 * @return {@link Status#READY} if there was only one event in the input
+	 *         list. Returns {@link Status#CALLBACK} if there were more than one
+	 *         events available in the list.
+	 * @throws HandlerException
+	 * 
+	 * @formatter:on
+	 */
+	@Override
+	public Status process() throws HandlerException {
+		logger.debug("processing handler", "processing avro handler");
```
Comment (a-ninja) [inline bigdime-data-handlers/src/main/java/io/bigdime/handler/json/JsonMapperHandler.java]: typo? avro?

File: bigdime-data-handlers/src/test/java/io/bigdime/handler/hive/AdapterMetaDataHandlerTest.java [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+import org.mockito.Mockito;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.testng.AbstractTestNGSpringContextTests;
+import org.springframework.test.util.ReflectionTestUtils;
+import org.testng.annotations.BeforeTest;
+import org.testng.annotations.Test;
+
+@ContextConfiguration(classes = { MetaDataJsonUtils.class})
+public class AdapterMetaDataHandlerTest extends AbstractTestNGSpringContextTests{
```
Comment (a-ninja) [inline bigdime-data-handlers/src/test/java/io/bigdime/handler/hive/AdapterMetaDataHandlerTest.java]: Adaptor or Adapter?

File: bigdime-data-handlers/src/test/java/io/bigdime/handler/hive/AdapterMetaDataHandlerTest.java [hunk 8]
```diff
@@ -0,0 +1,122 @@
...
+		headers.put(ActionEventHeaderConstants.HIVE_PARTITION_NAMES, "account,dt");
+		headers.put(ActionEventHeaderConstants.HIVE_PARTITION_VALUES, "testaccount,20150101");
+		headers.put(ActionEventHeaderConstants.HIVE_PARTITION_LOCATION, FileUtils.getTempDirectoryPath()+File.separator
+				+headers.get(ActionEventHeaderConstants.ENTITY_NAME)+File.separator+"20150101");
+
+		actionEvent.setHeaders(headers);
+		actionEvents.add(actionEvent);
+		handlerContext.setEventList(actionEvents);
+		adapterMetaDataHandler.process();
+	}
```
Comment (a-ninja) [inline bigdime-data-handlers/src/test/java/io/bigdime/handler/hive/AdapterMetaDataHandlerTest.java]: asserts?

File: bigdime-data-handlers/src/test/java/io/bigdime/handler/hive/AdapterMetaDataHandlerTest.java [hunk 9]
```diff
@@ -0,0 +1,122 @@
...
+
+	@Test
+	public void testBuild() throws JsonProcessingException, IOException, AdaptorConfigurationException, InterruptedException, MetadataAccessException{
+		AdapterMetaDataHandler adapterMetaDataHandler = mockAdapterMetaDataHandler();
+		adapterMetaDataHandler.setPropertyMap(mockProperties());
+		adapterMetaDataHandler.build();
+		adapterMetaDataHandler.build();
```
Comment (a-ninja) [inline bigdime-data-handlers/src/test/java/io/bigdime/handler/hive/AdapterMetaDataHandlerTest.java]: asserts?

File: bigdime-data-handlers/src/main/java/io/bigdime/handler/json/JsonMapperHandler.java (hunk 5 above)
Comment (mnamburi) [inline bigdime-data-handlers/src/main/java/io/bigdime/handler/json/JsonMapperHandler.java]: good catch

File: bigdime-commons/src/main/java/io/bigdime/core/commons/JsonHelper.java (hunk 2 above)
Comment (mnamburi) [inline bigdime-commons/src/main/java/io/bigdime/core/commons/JsonHelper.java]: I have pushed the test code base, will revert the changes.

**Comment by a-ninja** [comment]
ship it
